Add 2x5 shower-shape and saturation info for electrons and photons

rafaellopesdesa opened the pull request:
I am working to update #17101 , but I would like to have all showershapes used in the regression already calculated inside the objects. We recently decided to sync the electron and photon regressions and the electron object did not have the 2x5 shower shapes, which are being added. I also added the saturation information that is now added as input for the regression (in both GsfElectron and Photon. This is a smaller big change, because it has to be propagated to miniAOD, which is also done here.

Commit:
Add RPC readout links in CondFormats for RPC TwinMux/OMTF/CPPF unpackers

Commit:
Add analysers to fill RPC readout link maps in CondFormats (LB, DCC, TwinMux)

Commit:
Use FileInPath for RPCTwinMuxLinkMapInput.txt and remove obsolete .dat files from CondTools/RPC/data as mentioned in cms-sw/cmssw#16106

Commit:
Get coral session from pool and limit session and transaction scope in RPCDCCLinkMapHandler and RPCLBLinkMapHandler as mentioned in cms-sw/cmssw#16106

jfernan2 committed:
Changes to disable DDU trigger plots in DT DQM Online Trigger checks and disable check on RPC paths for TriggerEfficiency studies in DT DQM Offline. Port to 90X of PR 16679

jfernan2 committed:
Fix to solve runMatrix problem

jfernan2 committed:
Fix for special BeamHalo RelVal configuration

jfernan2 committed:
Changes in Digis and unpacker moved to EventFilter package

jfernan2 committed:
Modification for 2016 running Era

Commit:
new plots

Commit:
changes

Commit:
digi parameters

Commit:
implementing digi post processor

Commit:
implementing digi post processor

jfernan2 committed:
Final fix Run2 Era configuration preserving the previous data formats

Commit:
bkg rates

Commit:
adding sim segment

Commit:
adding matching to MC truth

Commit:
simToReco

Commit:
simToReco

Commit:
num seg eff

Commit:
fix num seg eff

Commit:
seg postvalidation

Commit:
fixes

Commit:
pu id + other changes

Commit:
fix digi efficiency

Commit:
fix delta time

Commit:
parameters for the simtrack selection

Commit:
bkg rate binning

Commit:
binning

Commit:
binning

Commit:
fix

ianna committed:
Add GEM payload producers to 2017 payload scripts

ianna committed:
Add GEM payload producers to 2017 payload scripts

Commit:
remove couts

jfernan2 committed:
Move from tabs to spaces

jfernan2 committed:
Another tab

Commit:
delete profile

jfernan2 committed:
Removed all tabs from python files in DQM/DT* packages

Commit:
fix LogWarning + title

Commit:
required fixes

Commit:
required fixes

Commit:
deleting cloned histos

depasse committed:
Reading TPG parameters from TPG_beamv6_trans_xxx_spikekill.txt file

depasse committed:
Reading TPG parameters from TPG_beamv6_trans_xxx_spikekill.txt file

Commit:
fix

depasse committed:
Reading TPG parameters from TPG_beamv6_trans_xxx_spikekill.txt file

jfernan2 committed:
Removed producers from source configuration cffs

makortel committed:
Use also TrackingParticle::numberOfHits()==0 in QuickTrackAssociatorByHits

In TrackingTruthAccumulator, the numberOfHits is calculated from a
subset of the SimHits of the SimTracks of a TrackingParticle
(essentially limiting the processType and particleType to those of the
"first" hit, and particleType to the pdgId of the SimTrack). But, here
the association between tracks and TrackingParticles is done with
*all* the hits of TrackingParticle, so we should not rely on the
numberOfHits() calculated with a subset of SimHits.

makortel committed:
Drop TrackingParticle::numberOfHits()==0 check also from the ntuple

makortel committed:
Sort SimHits by their time of flight for the number of hits calculation

Because the processType and particleType are taken from the "first"
SimHit of a SimTrack, it is important to get the "first" right.
Previously it was the one with the smallest index in the SimHit
vector, but this fails for Phase2 tracker where also the outer tracker
(OT) SimHits are in the same event products as pixel SimHits, and for
some reason the OT SimHits are inserted before the pixel SimHits.

I believe using the time of flight is more robust way to decide which
SimHit is "first".

makortel committed:
Small performance optimization

jhgoh committed:
Merged refs/pull/17252/head from repository cms-sw

jhgoh committed:
RPCRecHit x-position correction for endcap

ianna committed:
Add payload scripts for 2018 geometry scenario

ianna committed:
Update 2018 geometry scenario to pick up newest changes added to 2017

ianna committed:
Remove obsolete comment

rovere committed:
Silent warning due to missing templates in PhaseII

The materialDumpAnalyser uses internally a refitter to
extract the angle of incidence of the track with respect to
the sensor surface. The refitter uses 'angleAndTemplate',
that in PhaseII is missing. This PR drops the template
part.

manoni committed:
Implementation of several plots in SiStripMonitorCluster package: number of cluster trend in wheel/layer, number of cluster per ring and per layer. Change in apv charge median plot switch in SiStripMonitorDigi package

jhgoh committed:
seems to have flipped sign in x correction

VinInn committed:
use View in refitter

VinInn committed:
improve example and traj analyzer

lorusso7 committed:
Update to Madgraph 2.4.2

lorusso7 committed:
Update to Madgraph 2.4.2

lorusso7 committed:
Update to Madgraph 2.4.2

lorusso7 committed:
Update to Madgraph 2.4.2

knoepfel committed:
White space changes only.

knoepfel committed:
Remove unnecessarily specified d'tor.

knoepfel committed:
Convert LogErrorHarvester to global::EDProducer.

gartung committed:
Changes needed to compile with macOS clang and libc++

gartung committed:
Changes needed to compile with macOS clang and libc++

gartung committed:
Changes needed to compile with macOS clang and libc++

gartung committed:
Changes needed to compile with macOS clang and libc++

rafaellopesdesa committed:
add 2x5 shower-shape and saturation info for electrons and photons in miniAOD

davidlange6 committed:
remove comparisons of nonnull objects against 0

ggovi committed:
Removed CondCore/ORA package

cmsbuild commented:
A new Pull Request was created by  @rafaellopesdesa (Rafael Lopes de Sa) for CMSSW_9_0_X.

It involves the following packages:

DataFormats/EgammaCandidates
DataFormats/PatCandidates
PhysicsTools/PatAlgos
RecoEgamma/EgammaElectronAlgos
RecoEgamma/EgammaPhotonProducers


@cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please review it and eventually sign? Thanks.
@rappoccio, @Sam-Harper, @JyothsnaKomaragiri, @imarches, @ahinzmann, @acaudron, @mmarionncern, @TaiSakuma, @gpetruc, @lgray, @jdolen, @nhanvtran, @gouskos, @gkasieczka, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @cbernet this is something you requested to watch as well.
@davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

Dr15Jones committed:
The new gcc ABI for C++11 uses different namespace

The gcc ABI for C++11 internally uses the namespace std::__cxx11
for the standard library containers. The friendly class name
generator now knows about that naming in order to keep the
branch names the same as before the ABI change.

cmsbuild committed:
Merge pull request #17116 from calabria/NewME0Validation90X

New ME0 validation

cmsbuild committed:
Merge pull request #17249 from depasse/TPGfromFile

Reading TPG parameters from file

cmsbuild committed:
Merge pull request #17304 from manoni/CMSSW_9_0_ROOT6_X_2017-01-16-2300_clusTrend9bis

Modification/Addition in SiStripMonitorDigi and SiStripMonitorCluster packages

cmsbuild committed:
Merge pull request #17336 from lorusso7/Madgraph_2.4.2_cards

Madgraph 2.4.2 cards

cmsbuild committed:
Merge pull request #17338 from makortel/quickTrackAssocRemoveNumberOfHits

Remove check on TrackingParticle::numberOfHits()==0 from QuickTrackAssociatorByHits

cmsbuild committed:
Merge pull request #17354 from gartung/Fireworks-Core-macos-clang-fixes

Fireworks/Core : Changes needed to compile with macOS clang and libc++

cmsbuild committed:
Merge pull request #17290 from rovere/SilentTemplateWarnInMatDumper

Silent warning due to missing templates in PhaseII

cmsbuild committed:
Merge pull request #17355 from gartung/Geometry-CaloGeometry-macos-clang-fixes

Geometry/CaloGeometry : Changes needed to compile with macOS clang and libc++

davidlange6 committed:
Merge pull request #17366 from ggovi/condcore-remove-ora

Remove CondCore/ORA package

davidlange6 committed:
Merge pull request #17131 from ianna/gem-in-2017-payload-scripts

GEM Reco Geometry Payload Writer Added to 2017 Scripts

davidlange6 committed:
Merge pull request #17291 from ianna/extended-2018-geometry-scenario

Extended 2018 Geometry Scenario Update

cmsbuild committed:
Merge pull request #16739 from jfernan2/dropDDU_fromDTDQMtrigger_90X

Changes to disable DDU trigger plots in DT DQM Online Trigger checks …

davidlange6 committed:
Merge pull request #17359 from gartung/SimDataFormats-RandomEngineState-macos-clang-fixes

SimDataFormats/RandomEngineState : Changes needed to compile with macOS clang and libc++

Commit:
Update the gap value in GE21 design for TDR_Dev version

cmsbuild committed:
Merge pull request #16638 from fthyssen/rpc-amc-unpacker-conditions

RPC TwinMux and OMTF/CPPF unpacker conditions

cmsbuild commented:
-1

Tested at: 14d55b7c5cea21f3c54944827af6879dc4d512c4

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17370/17567/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
4.22 step4
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step4_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

</pre>135.4 step3
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step3_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>25.0 step5
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step5_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1001.0 step3
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step3_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>1000.0 step3
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step3_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

rafaellopesdesa commented:
Apparently the part of the code that fills the saturation info for photons is crashing. I didn't see that before, I apologize. I will run these RelVals locally and update with the appropriate fix asap.

gartung committed:
changes needed to compile on macOS sierra with system clang and libc++

gartung committed:
wrap change for rootcling

rafaellopesdesa committed:
classes dict and protection against null ptr

rafaellopesdesa committed:
Merged NewVarsElectronTmp from repository rafaellopesdesa

makortel committed:
Ignore upfront SimHits with particleType different from SimTrack pdgId

These are anyway ignored from the number of hit calculation, so let's
ignore them from the "first hit" deduction as well.

cmsbuild committed:
Merge pull request #17374 from bsunanda/Phase2-gem22

bsunanda:Phase2-gem22 Update the gap value in GE21 design for TDR_Dev version

cmsbuild committed:
Merge pull request #17377 from knoepfel/convert-LogErrorHarvester

Convert edm::LogErrorHarvester from legacy to global EDProducer.

cmsbuild committed:
Merge pull request #17342 from gartung/FWCore-Framework-macos-clang-fixes

FWCore/Framework : Changes needed to compile with macOS clang and libc++

cmsbuild committed:
Merge pull request #17371 from Dr15Jones/gccAbiChange

The new gcc ABI for C++11 uses different namespace

gartung committed:
add const

boudoul committed:
Removing cout from code used in prod

slava77 commented on DataFormats/EgammaCandidates/src/classes_def.xml:
```diff
@@ -220,8 +225,12 @@
     <version ClassVersion="2" checksum="3522539867"/>
   </class>
+  <class name="reco::GsfElectron::SaturationInfo" ClassVersion="0">
+   <version ClassVersion="0" checksum="3868678681"/>
```
version=0 is a special number.
start from 3

kpedro88 committed:
temp fix 2017 WF nums by skipping ranges

davidlange6 committed:
Revert "Pr90x L1T Stage2 for Phase2"

davidlange6 committed:
Merge pull request #17392 from cms-sw/revert-17248-pr90x-L1TStage2-forPhase2

Revert "Pr90x L1T Stage2 for Phase2"

rafaellopesdesa committed:
change version

rafaellopesdesa committed:
Merged NewVarsElectronTmp from repository rafaellopesdesa

cmsbuild commented:
Pull request #17370 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

rafaellopesdesa commented:
@slava77  Hi Slava, thanks a lot for the help with versions. I didn't know about this detail. The code is updated.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

davidlange6 committed:
Merge pull request #17391 from kpedro88/Phase1-WF2

temp fix 2017 WF nums by skipping ranges

cmsbuild commented:
+1
Tested at: 2339640b5ea8b2a4f8fbdbe88da4d1855da458d9
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17370/17593/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/2339640b5ea8b2a4f8fbdbe88da4d1855da458d9
https://github.com/cms-sw/cmssw/commit/c44ddbd3a922daa7a4a84276dcc67a3d39b237d4
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17370/17593/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17370/17593/git-merge-result

cmsbuild commented:
Comparison job queued.

davidlange6 committed:
Merge pull request #17390 from boudoul/silent

Removing 'cout' comments  from code used in prod (phase2 Track Trigger)

cmsbuild committed:
Merge pull request #17382 from makortel/fixTrackingParticleNumberOfHits

Fix calculation of TrackingParticle::numberOfHits()

cmsbuild committed:
Merge pull request #17380 from gartung/DataFormats-CaloTowers-macos-clang-fixes

DataFormats/CaloTowers : changes needed to compile on macOS sierra with system clang and libc++

cmsbuild committed:
Merge pull request #17364 from davidlange6/fix630warnings

remove comparisons of nonnull objects against 0

cmsbuild committed:
Merge pull request #17334 from VinInn/RefitWithView

use edm::View in TrackRefitter

cmsbuild committed:
Merge pull request #17289 from jhgoh/RPCRecHitXCorrection

RPCRecHit local x position correction

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17370/17593/summary.html

rafaellopesdesa commented:
I would like to use these new data members in the update of the PR for 90X regression... can I use it already?

slava77 commented:
On 2/7/17 9:57 AM, Rafael Lopes de Sa wrote:
> I would like to use these new data members in the update of the PR for
> 90X regression... can I use it already?
> 

Do you mean if you can combine this PR with the regression PR?
Sure, why not.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/17370#issuecomment-278084505>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcblPtXvkv0oCEUHiIXJmSM6dhFdnQks5raLBygaJpZM4Lz7ie>.
>

rafaellopesdesa commented:
I was wondering what would be easier. I have a meeting tomorrow with AlCA to request the new payloads in the global tag so that the regression passes validation (they requested a more detailed explanation that was easier to do talking). I was thinking to then, tomorrow, submit the update to that PR. If it is easier, I merge this one together myself...

But I can also wait for this to be merged in the cmssw repository, since this one already passed all the tests.

slava77 commented on DataFormats/EgammaCandidates/src/GsfElectron.cc:
```diff
@@ -71,6 +71,29 @@ GsfElectron::GsfElectron
+ ( int charge, const ChargeInfo & chargeInfo,
+   const GsfElectronCoreRef & core,
+   const TrackClusterMatching & tcm, const TrackExtrapolations & te,
+   const ClosestCtfTrack & ctfInfo,
```
ctfInfo appears unnecessary.
Please remove it from this constructor.
Older constructors kept it to preserve interface backward compatibility, which is not applicable here.

slava77 replied:
.. actually, do you even need to define this new constructor?
Can the old one be extended with one more argument?

rafaellopesdesa commented:
Yes, it can. I was trying to preserve the "logic" of the code (well, more
like style than logic). Should I just extend the latest and more complete
constructor?

-- Rafael.

On Wed, Feb 8, 2017 at 4:40 AM, Slava Krutelyov <notifications@github.com>
wrote:

> *@slava77* commented on this pull request.
> ------------------------------
>
> In DataFormats/EgammaCandidates/src/GsfElectron.cc
> <https://github.com/cms-sw/cmssw/pull/17370>:
>
> > @@ -71,6 +71,29 @@ GsfElectron::GsfElectron
>   }
>
>  GsfElectron::GsfElectron
> + ( int charge, const ChargeInfo & chargeInfo,
> +   const GsfElectronCoreRef & core,
> +   const TrackClusterMatching & tcm, const TrackExtrapolations & te,
> +   const ClosestCtfTrack & ctfInfo,
>
> .. actually, do you even need to define this new constructor?
> Can the old one be extended with one more argument?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/17370>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFtCpLDXxGAG_EDu189XLmHPMYEmpQy7ks5raZu2gaJpZM4Lz7ie>
> .
>

slava77 commented:
@rafaellopesdesa 
yes please,
just extend the latest most complete constructor (the one just above the proposed one)

rafaellopesdesa commented:
OK, will do later today. Thanks.

On Feb 9, 2017 1:56 AM, "Slava Krutelyov" <notifications@github.com> wrote:

> @rafaellopesdesa <https://github.com/rafaellopesdesa>
> yes please,
> just extend the latest most complete constructor (the one just above the
> proposed one)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/17370#issuecomment-278512990>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFtCpKuJcde9-e-MInaiATLFSq3Vml6Mks5ramQ5gaJpZM4Lz7ie>
> .
>

slava77 commented:
@rafaellopesdesa 
should I expect an update here soon.
Please clarify.
Thank you.

rafaellopesdesa commented:
Yes, I was in a conference these last days. Will update shortly.

Rafael

On Feb 10, 2017 6:20 PM, "Slava Krutelyov" <notifications@github.com> wrote:

> @rafaellopesdesa <https://github.com/rafaellopesdesa>
> should I expect an update here soon.
> Please clarify.
> Thank you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/17370#issuecomment-279005577>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFtCpGys2I2lhYx77pYhFLWmK_9dBO0Zks5rbJxWgaJpZM4Lz7ie>
> .
>

rafaellopesdesa committed:
Merged NewVarsElectron from repository rafaellopesdesa

rafaellopesdesa committed:
remove duplicate constructor as Slava requested and add setters as Sam requested

cmsbuild commented:
Pull request #17370 was updated. @ghellwig, @kpedro88, @fabozzi, @rekovic, @vanbesien, @govoni, @civanch, @monttj, @franzoni, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @cerminar, @slava77, @ggovi, @mmusich, @mulhearn, @arunhep, @perrozzi, @thuer, @dmitrijus, @srimanob, @hengne can you please check and sign again.

slava77 commented:
this needs a rebase
https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideHATS2016Git#CMSSW_rebasing_recipe

slava77 commented:
@rafaellopesdesa 
for a cleaner rebase, you will need to go back to the branch state before the merge ( 	de61ce5) and then follow the rebase instructions.

rafaellopesdesa commented:
I will close and do together with the regression... it's done, I will submit now.
